tests/test_merge.py

```diff
@@ -599,6 +600,14 @@ def test_merge_allow_objects(merge: Any) -> None:
     assert ret == {"a": 10, "foo": iv}
 
 
+def test_merge_with_allow_Dataframe() -> None:
+    kwargs = OmegaConf.create({"a": Dataframe()}, flags={"allow_objects": True})
+    cfg = OmegaConf.create()
+
+    cfg.merge_with(kwargs)
+    assert isinstance(cfg.a, Dataframe)
```

Comment:
If this still reproduces the problem, please switch to it:
```suggestion
def test_merge_with_allow_Dataframe() -> None:
    cfg = OmegaConf.create({"a": Dataframe()}, flags={"allow_objects": True})
    ret = OmegaConf.merge({}, cfg)
    assert isinstance(ret.a, Dataframe)
```